src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ActivityExtensions.cs

```diff
@@ -77,6 +77,11 @@ internal static class ActivityExtensions
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static void Return(this OtlpCollector.ExportTraceServiceRequest request)
         {
+            if (repeatedFieldOfSpanSetCountAction == null)
```

Comment:
We could probably add this check for `GetSpanListFromPool` as well to avoid taking the `ScopeSpan` from the pool if we are never going to return it back.

Reply:
GetSpanListFromPool still needs to be called as it is still the thing that news up a `ScopeSpans` object. It's just that if the pooling is broken in the future, GetSpanListFromPool simply will never find any `ScopeSpans` in the pool and always new up a new `ScopeSpans`.

Reply:
I meant that inside the `GetSpanListFromPool` method we could avoid the `ConcurrentBag.TryTake` call and always new up an `OtlpCommon.OtlpTrace.ScopeSpans` instance when reflection fails:

```diff
internal static OtlpTrace.ScopeSpans GetSpanListFromPool(string name, string version)
        {
+           if (repeatedFieldOfSpanSetCountAction == null || !SpanListPool.TryTake(out var spans))
            {
                spans = new OtlpTrace.ScopeSpans
                {
                    Scope = new OtlpCommon.InstrumentationScope
                    {
                        Name = name, // Name is enforced to not be null, but it can be empty.
                        Version = version ?? string.Empty, // NRE throw by proto
                    },
                };
            }
            else
            {
                spans.Scope.Name = name;
                spans.Scope.Version = version ?? string.Empty;
            }

            return spans;
        }
```

Reply:
Ah, sure I see what you're saying. I suppose I was thinking this stuff'll probably never break, but you're right it doesn't hurt to add the check.